### PR TITLE
Add OracleLinux 8 & 9 support

### PR DESCRIPTION
**Pull request by bastelfreak**
Currently testing in https://github.com/voxpupuli/puppet-yum/pull/319. Also requires https://github.com/voxpupuli/puppet_metadata/pull/98

### CODE REVIEW

File: lib/beaker-hostgenerator/data.rb [hunk 1]
```diff
@@ -1082,6 +1082,11 @@ def generate_osinfo
         yield ["oracle#{release}-64", "el-#{release}-x86_64"]
       end
 
+      # OracleLinux
```
Comment (ekohl) [inline lib/beaker-hostgenerator/data.rb:1085]: How is this different from oracle just above it?

Comment (bastelfreak) [reply lib/beaker-hostgenerator/data.rb:1085]: I was already wondering where that's coming from. We currently don't use that anywhere. Is that maybe a leftover that we can delete? Is Puppet using that internally for oracle builds?

Comment (bastelfreak) [reply lib/beaker-hostgenerator/data.rb:1085]: That's my assumption as well. We always use `OracleLinux` in metadata.json so I think we should use that here as well. I can raise another PR to remove the old code and we merge that when we do another major release.

Comment (ekohl) [reply lib/beaker-hostgenerator/data.rb:1085]: As I said in https://github.com/voxpupuli/beaker-hostgenerator/pull/322#pullrequestreview-1567195970, I'd like to continue using `oracle` as the name.

It avoids doing a major release (which is painful) and in daily operation it's also easier.

File: lib/beaker-hostgenerator/data.rb [hunk 2]
```diff
@@ -1090,6 +1090,11 @@ def generate_osinfo
...
+      (7..9).each do |release|
```
Comment (ekohl) [inline lib/beaker-hostgenerator/data.rb:1094]: This overlaps with line 1093. Can you merge the two blocks?

**Review by ekohl — APPROVED** [review]
Thanks!

**Pull request merged**